### PR TITLE
Fix product element styling class names

### LINKED ISSUE #3093: All Products Product Title text customization options only work if title is a link
## Describe the bug
The `Product Title` block [has new options for customizing text colour and size](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2860). These options only apply if the `Link to product page` option is enabled. If `Link to product page=false` then the title has default formatting and it can't be changed.

title customize only affects link

Found when testing 3.3.0 https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3077

## To reproduce
- Add an all products block (or edit existing)
- Click the `Edit` (pencil icon) button to edit the layout of products within the grid.
- Select the product title block, or add it if needed.
- In the settings inspector, change the text size and colour options.
- Click `Done` to exit layout edit.
- Disable `Link to product page` option in settings (sidebar).
- Publish the page or save changes.
- View the editor preview, and/or view page on front end.

Notice that the customization options only apply if link option is active. 

## Expected behavior
Customization options to work consistently when title is rendered as a plain (non link) heading. OR the options are not available when link mode is off.

### WordPress (please complete the following information):
* Core version: 5.5
* Gutenberg version (if installed): n/a
* WooCommerce version: 4.4.1
* WooCommerce Blocks version: 3.3.0-dev

### LINKED ISSUE #3094: All Products Product Title alignment options don't work
## Describe the bug
The `Product Title` block [has new options for aligning the text](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2860). The text is always center-aligned - the option doesn't seem to work. Similar issue with product price: #3049

Found when testing 3.3.0 https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3077

## To reproduce
- Add an all products block (or edit existing)
- Click the `Edit` (pencil icon) button to edit the layout of products within the grid.
- Select the product title block, or add it if needed.
- In the block toolbar, change the text alignment.
- Click `Done` to exit layout edit, then publish the page.
- View the editor preview, and/or view page on front end.

Notice that the alignment option only works in layout editor - text is centered in editor preview and front end.

### WordPress (please complete the following information):
* Core version: 5.5
* Gutenberg version (if installed): n/a
* WooCommerce version: 4.4.1
* WooCommerce Blocks version: 3.3.0-dev

### PR DESCRIPTION
**Pull request by Aljullu**
Fixes #3093.
Fixes #3094.

### Screenshots

Peek 2020-09-01 12-13

### How to test the changes in this Pull Request:

1. Create a page with the _All Products_ block.
2. Edit it and click on the Product title.
3. In the sidebar, change the color and font size. In the toolbar, change the alignment.
4. Verify all styles are applied correctly.
5. Toggle the 'Link to Product Page' attribute and verify styles are still applied.
6. Click on 'Update' and preview the page in the frontend.
6. Verify styles are applied there as well.

It would be good to test two additional blocks to make sure there are no regressions:
1. While editing the _All Products_ block, click on the Product price.
2. In the sidebar, change the colors and font size. In the toolbar, change the alignment.
3. Verify all styles are applied correctly.
4. Now click on the product image block and change the 'Sale Badge Alignment' attribute.
5. Verify the 'Sale Badge' changes its position according to the attribute.
7. Click on 'Update' and preview the page in the frontend.
8. Verify styles are applied there as well.

cc @haszari I'm adding this to the 3.3.0 milestone in case you want to include it, but feel free to move it to 3.4.0 if you don't want to rush this fix.

### Changelog

> Fixed styling options of the Product Title block.

### CODE REVIEW

File: assets/js/atomic/blocks/product-elements/price/style.scss [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+.wc-block-components-product-price--align-left {
 	text-align: left;
 }
-.wc-block-components-product-price__align-center {
+.wc-block-components-product-price--align-center {
 	text-align: center;
 }
-.wc-block-components-product-price__align-right {
+.wc-block-components-product-price--align-right {
 	text-align: right;
```
Comment (senadir) [inline assets/js/atomic/blocks/product-elements/price/style.scss:9]: this change is unrelated to the issue, right? just styling fixes?

**Review by senadir — APPROVED** [review]
works like a sharm.

File: assets/js/atomic/blocks/product-elements/price/style.scss (hunk 1 above)
Comment (Aljullu) [reply assets/js/atomic/blocks/product-elements/price/style.scss:9]: Yes, I wanted to align all class names to follow the same pattern (`--` instead of `__`) even in blocks where there was not an issue.

**Comment by haszari** [comment]
> cc @haszari I'm adding this to the 3.3.0 milestone in case you want to include it, but feel free to move it to 3.4.0 if you don't want to rush this fix.

Thanks @Aljullu  - this seems like a fairly safe, robust change, so I'm going to include it. I'll cherry pick into the release branch and do another round of testing with the built plugin.

When reviewing the code in this PR and the element blocks, I noticed that some of the styling is hidden behind feature flag. So I'm keen to do a good round of testing with release build to check behaviour is as expected.

Examples:

- https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/cdc95c56f6aa5942414d60fa5232fb674b71b42f/assets/js/atomic/blocks/product-elements/title/block.js#L75
- https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/cdc95c56f6aa5942414d60fa5232fb674b71b42f/assets/js/atomic/blocks/product-elements/title/block.js#L70

**Comment by haszari** [comment]
I found one regression/issue (maybe preexisting) in Twenty Twenty when testing this, logged a follow up issue: #3097
